### PR TITLE
cleanup + github action + php 8.1

### PR DESCRIPTION
**Pull request by Kanti**

### CODE REVIEW

File: src/Tools/EnvironmentUtility.php [hunk 1]
```diff
@@ -100,6 +101,9 @@ public function getInstalledPackageVersion(string $packageName): ?string
     {
         $contents = file_get_contents($this->spotVendorPath() . '/composer/installed.json') ?: '';
         $activeInstallations = json_decode($contents, false);
+        if (!($activeInstallations instanceof stdClass)) {
```
Comment (ChrisB9) [inline src/Tools/EnvironmentUtility.php:104]: under what circumstance will this happen?

File: tests/Config/TempSshConfigFileTest.php [hunk 2]
```diff
@@ -23,25 +24,24 @@ public function setUp(): void
...
-//        try {
-//            new TempSshConfigFile();
-//        } finally {
-//            $property->setValue($oldValue);
-//        }
+        $reflection = new \ReflectionClass(TempSshConfigFile::class);
+        $property = $reflection->getProperty('fileName');
+        $property->setAccessible(true);
+        $oldValue = $property->getValue();
+        $property->setValue('/etc/hosts/.phpsu/ssh_config');
```
Comment (ChrisB9) [inline tests/Config/TempSshConfigFileTest.php:37]: why specifically this path?

File: tests/Helper/ApplicationHelperTest.php [hunk 3]
```diff
@@ -24,6 +24,7 @@ public function testGetPhpSuVersionFromGitFolder(): void
     {
         $versionString = $this->callPrivateMethod('getPhpSuVersionFromGitFolder', self::GIT_PATH);
         $this->assertIsString($versionString);
+        assert(is_string($versionString)); # dump phpstan :(
```
Comment (ChrisB9) [inline tests/Helper/ApplicationHelperTest.php:27]: `dumb` 
I apparently can't add suggestions here...

Comment (ChrisB9) [reply tests/Helper/ApplicationHelperTest.php:27]:
```suggestion
        assert(is_string($versionString)); # dumb phpstan :(
```

File: tests/phpunit.xml [hunk 4]
```diff
@@ -15,22 +17,24 @@
     timeoutForMediumTests="1"
     timeoutForSmallTests="1"
     colors="true">
+  <coverage>
+    <include>
+      <directory suffix=".php">../src/</directory>
+    </include>
+    <report>
+      <clover outputFile="test-results/coverage.xml"/>
```
Comment (ChrisB9) [inline tests/phpunit.xml:25]: :+1: i like this

File: .editorconfig [hunk 5]
```diff
@@ -0,0 +1,654 @@
+root = true
```
Comment (ChrisB9) [inline .editorconfig:1]: This file contains a lot of defaults, doesn't it?

File: src/Tools/EnvironmentUtility.php (hunk 1 above)
Comment (Kanti) [reply src/Tools/EnvironmentUtility.php:104]: phpstan

File: tests/Config/TempSshConfigFileTest.php (hunk 2 above)
Comment (Kanti) [reply tests/Config/TempSshConfigFileTest.php:37]: this is always a file and never a directory. => will always result in an error

File: .editorconfig (hunk 5 above)
Comment (Kanti) [reply .editorconfig:1]: Better that there are default settings and we change them along the way.
Than none at all.

**Pull request merged**